src/Indicator.vala

```diff
@@ -82,15 +81,25 @@ public class DateTime.Indicator : Wingpanel.Indicator {
         return main_grid;
     }
 
+    private void create_event_grid () {
+        event_grid = new Gtk.Grid ();
+        event_grid.orientation = Gtk.Orientation.VERTICAL;
+        main_grid.attach (event_grid, 0, 1);
+    }
+
     private void update_events_model (E.Source source, Gee.Collection<E.CalComponent> events) {
         Idle.add (update_events);
     }
 
     private bool update_events () {
-        foreach (var w in event_box.get_children ()) {
-            w.destroy ();
-        }
-        foreach (var e in Widgets.CalendarModel.get_default ().get_events (calendar.selected_date)) {
+        event_grid.destroy();
```

Comment:
You need to check event_grid exists (non-null) here.  If you move between days with no events twice in a row it will be null on the second function call.